server/src/test/java/io/crate/planner/operators/JoinTest.java

```diff
@@ -309,16 +324,18 @@ public void testHashJoin_TablesSwitchWhenRightBiggerThanLeft() {
         rowCountByTable.put(TEST_DOC_LOCATIONS_TABLE_IDENT, new Stats(100, 0, Map.of()));
         e.updateTableStats(rowCountByTable);
 
-        LogicalPlan eval = createLogicalPlan(mss);
-        assertThat(eval).isExactlyInstanceOf(Eval.class);
-        LogicalPlan hashjoin = eval.sources().get(0);
+        LogicalPlan plan = buildLogicalPlan(mss);
+        assertThat(plan).isEqualTo("Eval[name, id]\n" +
+                                   "  └ Eval[name, id, id]\n" +
+                                   "    └ HashJoin[(id = id)]\n" +
+                                   "      ├ Collect[doc.locations | [id] | true]\n" +
+                                   "      └ Collect[doc.users | [name, id] | true]");
+
+        LogicalPlan hashjoin = plan.sources().get(0).sources().get(0);
         assertThat(hashjoin).isExactlyInstanceOf(HashJoin.class);
         assertThat(((HashJoin) hashjoin).lhs().getRelationNames())
             .as("Smaller table must be on the left-hand-side")
             .containsExactly(TEST_DOC_LOCATIONS_TABLE_IDENT);
-
-        Join join = buildJoin(hashjoin);
```

Comment:
Why did you remove this? not revalant?

Reply:
The table swap used to be in the build method of hashjoin, but since now everything happens on logical plan level already there is no need anymore to check for the execution plan.

Reply:
i could probably even remove more code.

Reply:
i think for most cases the logical plan comparision would be enough.